info/CALCULATION-ru.md

```diff
@@ -0,0 +1,190 @@
+    const diff = cycleLength - currentDay;
+
+    if (diff > 14){
+        return `Ovulation in ${diff - 14} Days`;
+    }
+    if (diff === 14){
+        return "Ovulation today";
+    }
+    if (diff >= 12){
+        "Ovulation possible"
+    }
+    return "Ovulation finished";
+}
+```
+
+## Шанс забеременеть
+
+Как написано выше, беременность происходит только в овуляцию. Тогда получается, что в дни овуляции высокий шанс забеременеть. Тогда функция будет выглядеть так:
+
+```ts
+function getPregnancyChance(startDate: Date, cycleLength: number){
+    const currentDay = getCurrentDay(startDate, cycleLength);
+    const diff = cycleLength - currentDay;
+
+    if (diff === 14){
+        return "High";
+    }
+    return "Low";
+}
+```
+
+Но, как описано в предыдущем разделе, овуляция может происходить не только в день расчетов. Поэтому добавим равную в 2 дня погрешность:
+
+```ts
+function getPregnancyChance(startDate: Date, cycleLength: number){
+    const error = 2;
+    const currentDay = getCurrentDay(startDate, cycleLength);
+    const diff = cycleLength - currentDay;
+
+    if (diff >= 14 - error && diff <= 14 + error){
+        return "High";
+    }
+    return "Low";
+}
+```
+
+## Количество дней до месячных
+
+Чтобы узнать, сколько дней осталось до месячных, нам необходимо знать два значения: средняя длина цикла ```lengthCycle``` и дата начала текущего цикла ```startDate```. Тогда расчеты будут выглядеть так:
+
+```ts
+function getDaysBeforePeriod(startDate: Date, cycleLength: number){
+    const currentDay = getCurrentDay(startDate, cycleLength);
+    const diff = cyclesLength - currentDay;
+    if (diff === 0){
+        return "Period is today";
+    }
+    return `Period in ${diff} Days`;
+}
+```
+
+Может возникнуть ситуация, когда предполагаемый день месячный прошел, а месячные так и не наступили. В этом случае введем понятие "задержка". Задержка до 8-10 дней считается нормой, больше может говорить о некоторых проблемах со здоровьем, беременности и другом.
+
+После этого функция будет выглядеть так:
+
+```ts
+function getDaysBeforePeriod(startDate: Date, cycleLength: number){
+    const currentDay = getCurrentDay(startDate, cycleLength);
+    const diff = cyclesLength - currentDay;
+    if (diff > 0){
+        return `Period in ${diff} Days`;
+    }
+    if (diff === 0){
+        return "Period is today";
+    }
+    return `Delay ${Math.abs(diff)} Days`;
+}
+```
+
+## Фазы цикла
+
+### Теоретическая информация
+
+Весь менструальный цикл можно разделить на 2 фазы: фолликулярную и лютеиновую. Фолликулярна фаза начинается с первого дня месячных и заканчивается овуляцией. Лютеиновая фаза начинается после овуляции и заканчивается месячными. В течение этих фаз главные гормоны постоянно перестраиваются, что сказывается на самочувствие. Ниже на картинке представлен график 28-дневного цикла. На нем показано, как меняются гормоны в каждую фазу.
+
+![Фазы менструального цикла](https://qpicture.ru/images/2022/12/27/crc.jpg)

```

Comment:
The image from remote server may be deleted. Let's sage it and use our local copy

Reply:
I'd like to insert my own image in the future.
So I don't want to save these images. If you save them, you need to write links.